lib/pbench/server/api/resources/users_api.py

```diff
@@ -281,10 +281,21 @@ def post(self):
         }
         return make_response(jsonify(response_object), 200)
 
+
+class Logout(Resource):
+    """
+    Pbench API for User logout and deleting an auth token
+    """
+
+    def __init__(self, config, logger, auth):
+        self.server_config = config
+        self.logger = logger
+        self.auth = auth
+
     @Auth.token_auth.login_required(f=Auth().verify_auth())
     def delete(self):
```

Comment:
Yeah flask views are much more handy in this scenarios

Reply:
Well, more important than implementation quirks, I really like following the HTTP REST model treating URIs as resource identifiers (ideally opaque) and HTTP methods as CRUD operations on the resources. It's simple and comprehensive. Thus, a user is a resource: you create it either by `POST /users` receiving a generated id in the response or by `PUT /users/<id>`; query with `GET /users/<id>` (or `GET /users` if you're privileged to get all users), you edit with `PUT /users/<id>` and you delete with `DELETE /users/<id>`.

An authorization session would work the same way with something like `/authorization/<id>` so you'd "log in" by `POST /authorization` (edit isn't meaningful) and "log out" with `DELETE /authorization/<id>`.

Whereas a "functional" interface would use `POST` or `PUT` where you have data to pass, and `GET` where you don't, the URI names would be operations rather than resources, like you've got with `/login` and `/register`. It works, and it's certainly been done, but I don't like it as well; the interface tends to be more complicated and less logical, partly because there are generally relatively few resources with well-defined HTTP operations... which flip into awkwardly named functional URIs with less clearly defined operations.

Either works, and my preferences have been swept over the waterfall here; I'll survive.

But the real point is that if we're going to go with the more complex and less logical "functional" style, let's at least keep the HTTP methods consistent; DELETE, in particular, makes no sense here; we should consistently use POST or PUT.

Reply:
Yes I agree since the login and logout are two different resources there is no need to have `DELETE` protocol on logout. I updated it to `POST`